rust/scx_utils/src/topology.rs

```diff
@@ -139,6 +140,8 @@ pub struct Core {
     /// Ancestor IDs.
     pub llc_id: usize,
     pub node_id: usize,
+    /// A monotonically increasing id for each CPU that is unique.
+    pub hw_id: usize,
```

Comment:
Let's make `core.id` monotonically increasing and `core.hw_id` the kernel provided one. Maybe `core.kerrnel_id` is a better name.

Reply:
Sounds good.